### ISSUE TITLE
Widget

### ISSUE THREAD

**Issue by JediBrooker** [TheAlienDrew/Unofficial-Sticky-Notes#1]
Please create a widget.

**Comment by TheAlienDrew** [comment]
I've already been researching into what I can do about this. Since I still have college and work to do, I won't be able to get to this for a while. Sorry for the inconvenience. But it is still planned.

**Comment by wingers999** [comment]
+1 for widget - that would be fantastic since Microsoft still can't sort out having Sticky Notes in OneNote on a tablet device.  Great work !!!

**Comment by TheAlienDrew** [comment]
For anyone wanting to know my progress on this, I've gotten the main note extraction out of the way. I'm just now trying to work out how I should be creating the widget with correct formatting.

**Comment by TheAlienDrew** [comment]
More progress made, almost ready to release the update, just trying to work out a few issues with scrolling the notes, and bullet forrmatting.

**Comment by TheAlienDrew** [comment]
Unless it's really important, I've got everything working now, except for scrolling and note auto syncing (to widget).

I'll probably get some rest, and after that, I'll upload the app into the beta testing part of Google Play, so those that want to test it out with how it's coming so far, can so do.

**Comment by TheAlienDrew** [comment]
Anyways, I've submitted it to the store, when it becomes available, there will be an option to opt into the beta release, that's what you'll want to do if you want to test the current build.

I'm already thinking about trying to get 2 more features added too, but they rank lower priority for now, but I'm thinking about getting the widget, when clicked on, to open backup the note it came from in the app, and a separate features of using 'pinch-to-zoom' like gestures to increase/decrease the text size on the notes since I know everyone has their particular preferences.

As for feedback, I suggest commenting here, since I'm more apt to see github responses that anything coming from google play.

**Comment by wingers999** [comment]
Excellent, look forward to taking a look

**Comment by wingers999** [comment]
When will I see the option to opt into beta?

**Comment by TheAlienDrew** [comment]
It will appear on the Google page, you may need to scroll down a bit, but
it will say something around the lines of "Join the beta to get access..."

>

**Comment by wingers999** [comment]
Okay thanks, I assume just a delay between you submitting it and it appearing

**Comment by TheAlienDrew** [comment]
> Okay thanks, I assume just a delay between you submitting it and it appearing

Maybe, I suggest reloading the Play Store in case you haven't already.

**Comment by wingers999** [comment]
Yes done that, still shows last update as January 12th

**Comment by TheAlienDrew** [comment]
So even if you scroll all the way down, on the Google app page, you don't see this? ![Screenshot_20210128-073954_Google Play Store.jpg](https://user-images.githubusercontent.com/19640692/106161619-14cf6400-613c-11eb-9d6e-d721a38823a8.jpg)

**Comment by wingers999** [comment]
No I don't - this is page I am going to when checking from my desktop PC - https://play.google.com/store/apps/details?id=aliendrew.ms.stickynotes - will go and check store app on tablet shortly in case it appears on there

**Comment by TheAlienDrew** [comment]
> No I don't - this is page I am going to when checking from my desktop PC - https://play.google.com/store/apps/details?id=aliendrew.ms.stickynotes - will go and check store app on tablet shortly in case it appears on there

Oh yeah then it definitely won't show up if you're viewing it from a computer. Must be viewing it from the actual Play Store on the device that you installed it on.

**Comment by wingers999** [comment]
Sorted, now installed.  Initial comment is it looks good, but point of having it on main screen as widget would be to also be able to click on it and type / edit the notes?  or in future to use pen for inking

**Comment by TheAlienDrew** [comment]
> Sorted, now installed.  Initial comment is it looks good, but point of having it on main screen as widget would be to also be able to click on it and type / edit the notes?  or in future to use pen for inking

Unfortunately due to how the app works, making it so you can edit notes from the widget wouldn't really be possible, same for the ink. However from within the app, I'm looking into the idea of getting ink to work, but I have no inked notes on any of my own accounts, and not sure how to create one from within the desktop app in order to do testing with it.

**Comment by wingers999** [comment]
If you install Microsoft Launcher on a Android device then you can click New > Ink Note to create one, and if you haven't got a stylus you can just draw with finger.

I understand re not beng able to edit in widget, but could we have it so when you click on a note widget it opens it open in the main sticky notes app?

**Comment by TheAlienDrew** [comment]
Yeah I plan to do that with the notes soon. If I can ever fully uncover the
web's API.

>

**Comment by wingers999** [comment]
okay, post here when anything new to try and I will test and give feedback for you - thanks for working on it

**Comment by wingers999** [comment]
noticed one issue - I had deleted the note which was showing as widget from my main computer but it still shows on widget even though that note no longer exists

**Comment by wingers999** [comment]
would also be handy if you could long press on the note and then choose another note to show instead, rather than having to remove it and then add widget again

**Comment by TheAlienDrew** [comment]
> noticed one issue - I had deleted the note which was showing as widget from my main computer but it still shows on widget even though that note no longer exists

Yes I know about this. Part of the reason why is because I have an implemented note sync yet. However when I do, under Android widget documentation, I won't be able to progmatically remove the note, or at least so I understand.
